pkg/auth/providers/saml/saml_provider.go

```
@@ -192,14 +256,29 @@ func (s *Provider) saveSamlConfig(config *v3.SamlConfig) error {
 	}
 
 	config.SpKey = common.GetName(config.Type, field)
+	if s.name == ShibbolethName {
+		samlConfig := v3.SamlConfig{}
+		config.DeepCopyInto(&samlConfig)
 
-	logrus.Debugf("updating samlConfig %s", s.name)
-	_, err = s.authConfigs.ObjectClient().Update(config.ObjectMeta.Name, config)
-	if err != nil {
+		ldapConfig, certpool, err := s.getLDAPConfig()
+		if err != nil {
+			return err
+		}
+
+		ldapConfigCopy := v3.LdapConfig{}
+		ldapConfig.DeepCopyInto(&ldapConfigCopy)
+		s.hasValidLdapConfig = ldaputils.ValidateLdapConfig(ldapConfigCopy, certpool)
```

are you sure you don't want to return an error here?